addons/pos_self_order/static/src/self_order_service.js

```diff
@@ -134,25 +141,41 @@ export class SelfOrder {
     }
 
     async getOrdersFromServer() {
-        const accessTokens = this.orders.map((order) => order.access_token);
+        const accessTokens = this.orders.map((order) => order.access_token).filter(Boolean);
+
+        if (accessTokens.length === 0) {
+            return;
+        }
 
         try {
-            const orderChanges = {};
             const orders = await this.rpc(`/pos-self-order/get-orders/`, {
                 access_tokens: accessTokens,
             });
 
-            this.orders = this.orders.filter((order) => {
-                orderChanges[order.access_token] = order.lastChangesSent;
-                return !order.access_token;
-            });
+            const notSendData = this.orders
+                .filter((o) => o.access_token)
+                .reduce((acc, order) => {
+                    acc[order.access_token] = {
+                        lastChangesSent: order.lastChangesSent,
+                        lines: order.lines.filter((l) => !l.id),
+                    };
+                    return acc;
+                }, {});
```

Comment:
If we can avoid using reduce, we should. It's not a very intuitive function to use. For this one, I'd like to refactor this to:

```js
const notSendData = Object.fromEntries(this.orders.filter(o => o.access_token).map(o => ([o.access_token, {lastChangesSent: o.lastChangesSent, lines: o.lines.filter(l => !l.id)}])))
```

But if you disagree with the suggestion, you can still keep the pattern without using `reduce`:

```js
const notSendData = {};
for (const order of this.orders.filter(o => o.access_token)) {
  notSendData[order.access_token] = {
                        lastChangesSent: order.lastChangesSent,
                        lines: order.lines.filter((l) => !l.id),
                    }
}
```

One more thing, should we rename it to: `notSendData -> ordersToRecreate`?

Reply:
I agree with that this will be done after rebasing on this PR: https://github.com/odoo/odoo/pull/125743